diff --git a/.npmignore b/.npmignore
deleted file mode 100644
index 391688ff93de1791aa336f1d3c75a10a2ee60b7d..0000000000000000000000000000000000000000
diff --git a/hammer.js b/hammer.js
index 00d2965ebdb42c4ef17b2258fc5402cd26279266..8b13aae1cc3978cffeee3cf4f9c35ad49ac56fd7 100644
--- a/hammer.js
+++ b/hammer.js
@@ -2640,4 +2640,4 @@ if (typeof define === 'function' && define.amd) {
     window[exportName] = Hammer;
 }
 
-})(window, document, 'Hammer');
+})(typeof window === 'object' ? window : {}, typeof document === 'object' ? document : {createElement: () => ({style: {}})}, 'Hammer');
